int32 transporter_num
TransporterTask[] transporter_tasks